From 1437e4e5b405f04e1b3900fe84d23933ae66b351 Mon Sep 17 00:00:00 2001
From: Neti Ravi Kumar <ravineti@codeaurora.org>
Date: Thu, 13 Dec 2012 15:31:20 +0530
Subject: [PATCH] Revert "hardware/qcom/display: Remove glfinish call"

This reverts commit 95bc6a5c6dd76a432baaa6554abd0f4f7ca481f1.

Without glFinish(), in few corner cases; application is trying to
lock the buffer for write, while the read lock is still held.

CRs-Fixed: 430212

Change-Id: I9e3a22ca3496ef42e8dce5d565bcfe79fd427a4d
---
 libgralloc/framebuffer.cpp    |    3 +--
 libhwcomposer/Android.mk      |    2 +-
 libhwcomposer/hwc_copybit.cpp |    9 ---------
 3 files changed, 2 insertions(+), 12 deletions(-)

diff --git a/libgralloc/framebuffer.cpp b/libgralloc/framebuffer.cpp
index 517b92c..5531d70 100644
--- a/libgralloc/framebuffer.cpp
+++ b/libgralloc/framebuffer.cpp
@@ -152,9 +152,8 @@ static int fb_post(struct framebuffer_device_t* dev, buffer_handle_t buffer)
 static int fb_compositionComplete(struct framebuffer_device_t* dev)
 {
     // TODO: Properly implement composition complete callback
-#ifdef ANCIENT_GL
     glFinish();
-#endif
+
     return 0;
 }
 
diff --git a/libhwcomposer/Android.mk b/libhwcomposer/Android.mk
index b7a1b5f..c9a0a51 100644
--- a/libhwcomposer/Android.mk
+++ b/libhwcomposer/Android.mk
@@ -7,7 +7,7 @@ LOCAL_MODULE_TAGS             := optional
 LOCAL_C_INCLUDES              := $(common_includes)
 LOCAL_SHARED_LIBRARIES        := $(common_libs) libEGL liboverlay libgenlock \
                                  libhwcexternal libqdutils libhardware_legacy \
-                                 libdl libmemalloc libhwcservice libGLESv1_CM
+                                 libdl libmemalloc libhwcservice
 
 LOCAL_CFLAGS                  := $(common_flags) -DLOG_TAG=\"hwcomposer\"
 LOCAL_ADDITIONAL_DEPENDENCIES := $(common_deps)
diff --git a/libhwcomposer/hwc_copybit.cpp b/libhwcomposer/hwc_copybit.cpp
index 76f7852..c17fa7e 100644
--- a/libhwcomposer/hwc_copybit.cpp
+++ b/libhwcomposer/hwc_copybit.cpp
@@ -21,7 +21,6 @@
 #define DEBUG_COPYBIT 0
 #include <copybit.h>
 #include <genlock.h>
-#include <GLES/gl.h>
 #include "hwc_copybit.h"
 #include "comptype.h"
 #include "egl_handles.h"
@@ -188,14 +187,6 @@ bool CopyBit::draw(hwc_context_t *ctx, hwc_layer_list_t *list, EGLDisplay dpy,
         return -1;
     }
 
-#ifndef ANCIENT_GL
-    // Invoke a glFinish if we are rendering any layers using copybit.
-    // We call glFinish instead of locking the renderBuffer because the
-    // GPU could take longer than the genlock timeout value to complete
-    // rendering
-    glFinish();
-#endif
-
     for (size_t i=0; i<list->numHwLayers; i++) {
         if (list->hwLayers[i].compositionType == HWC_USE_COPYBIT) {
             retVal = drawLayerUsingCopybit(ctx, &(list->hwLayers[i]),
-- 
1.7.10.4

